### PR TITLE
Allow XDS v2/v3 for status distribution

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/19885

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/eventhandler.go [hunk 1]
```diff
@@ -14,11 +14,47 @@
...
+}
+
 // EventHandler allows for generic monitoring of xDS ACKS and disconnects, for the purpose of tracking
 // Config distribution through the mesh.
 type DistributionStatusCache interface {
 	// RegisterEvent notifies the implementer of an xDS ACK, and must be non-blocking
-	RegisterEvent(conID string, xdsType string, nonce string)
-	RegisterDisconnect(s string, urls []string)
-	QueryLastNonce(conID string, xdsType string) (noncePrefix string)
+	RegisterEvent(conID string, distributionType DistributionType, nonce string)
```
Comment (ramaraochavali) [inline pilot/pkg/proxy/envoy/v2/eventhandler.go:57]: nit: `EventType` seems more meaningful in the context  of method call `RegisterEvent`?

File: pilot/pkg/status/reporter.go [hunk 2]
```diff
@@ -65,6 +65,8 @@ type Reporter struct {
 	distributionEventQueue chan distributionEvent
 }
 
+var _ v2.DistributionStatusCache = &Reporter{}
```
Comment (therealmitchconnors) [inline pilot/pkg/status/reporter.go:68]: I'm curious, what does this line do?

Comment (howardjohn) [reply pilot/pkg/status/reporter.go:68]: it asserts Reporter implements DistributionStatusCache

File: pilot/pkg/proxy/envoy/v2/eventhandler.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/proxy/envoy/v2/eventhandler.go:57]: updated

File: pilot/pkg/proxy/envoy/v2/eventhandler.go [hunk 3]
```diff
@@ -14,11 +14,47 @@
...
+
+const (
+	ClusterEventType  EventType = "Cluster"
+	ListenerEventType EventType = "Listener"
+	RouteEventType    EventType = "Route"
+	EndpointEventType EventType = "Endpoint"
+	UnknownEventType  EventType = ""
+)
+
+var AllDistributionTypes = []EventType{
```
Comment (ramaraochavali) [inline pilot/pkg/proxy/envoy/v2/eventhandler.go:31]: nit: This needs to be changed as well?

**Pull request merged**